### DISCUSSION TITLE
Cypress Testing of application

### DISCUSSION THREAD

**Discussion by heemankv** [WalletConnect/walletconnect-monorepo#2430] [Developer Support] [answered]
I have a next js website that has web3modal as the web3 connection support, 
I am trying cypress automation testing to connect my website to metamask using web3modal

( it is to note that I am using synpress, a cypress fork that allows to play with  metamask-extension )
(https://github.com/Synthetixio/synpress)

I am having issues in capturing the web3modal elements in my cypress script
for eg
<img width="350" alt="Screenshot 2023-05-20 at 12 14 52 PM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/76938871/5fbf9301-cd1b-4880-a8bf-f7906d6b479a">

I am unable to catch this desktop title using this script
<img width="546" alt="Screenshot 2023-05-20 at 12 15 35 PM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/76938871/5639949b-37ab-4218-8ba5-268cc3adaa25">
<img width="383" alt="Screenshot 2023-05-20 at 12 17 54 PM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/76938871/ee99e1f9-a03f-4088-8626-47a4a29a5b99">

It is to note that I have already tried to capture various other self created modals and I am able to access their content, but it's just Cypress is unable to detect web3modal modal 

Please let me know what can be done, Thanks a lot in advance
WAGMI !!!

**Comment by finessevanes** [comment]
Hey - this has seen and has been shared w/ our team.

**Comment by chris13524** [comment, marked as answer]
I have not used Cypress myself, but it's possible you need to traverse inside of the shadow DOM since our components rely heavily on shadow DOMs. (Note the #shadow-root tags in the developer view.) These docs may help: https://docs.cypress.io/api/commands/shadow

We are actually currently working on [something similar with Playwright](https://github.com/WalletConnect/web3modal/blob/feat/playwright/laboratory/tests/wallet.spec.ts#L16), but Playwright will traverse the shadow DOM [automatically](https://playwright.dev/docs/locators#locate-in-shadow-dom).
